### PR TITLE
Command linker properties

### PR DESCRIPTION
**Pull request by ian-r-rose**
Implements command linker through `AttachedProperties` rather than DOM data attributes.

cc. @jasongrout @afshin

### CODE REVIEW

File: packages/apputils/src/commandlinker.ts [hunk 1]
```diff
@@ -97,10 +79,8 @@ export class CommandLinker implements IDisposable {
    * using the `populateVNodeDataset` method in order to disconnect them from
    * executing their command/argument pair.
    */
-  disconnectNode(node: HTMLElement): HTMLElement {
-    node.removeAttribute(`data-${COMMAND_ATTR}`);
-    node.removeAttribute(`data-${ARGS_ATTR}`);
-    return node;
+  disconnectNode(node: HTMLElement): void {
+    AttachedProperty.clearData(node);
```
Comment (jasongrout) [inline packages/apputils/src/commandlinker.ts]: We don't want to clear *all* property data, just the linkedCommandProperty data.

**Comment by blink1073** [comment]
I think the fact that it doesn't work for VDOM is enough reason not to make the change.

File: packages/apputils/src/commandlinker.ts (hunk 1 above)
Comment (ian-r-rose) [reply packages/apputils/src/commandlinker.ts]: I agree, but I don't see any `clear` function on `AttachedProperty`. Maybe a `WeakMap` would be the better choice.

**Comment by ian-r-rose** [comment]
While this is not properly a security hole, since it only works with trusted HTML, it does seem like this level of access to JupyterLab is not really intended by the command linker. For instance, I was able to use it to make a button that, when clicked, removed arbitrary files from the server.

**Comment by ian-r-rose** [comment]
Closing for now. If we want to re-open the discussion, let me know. It's also possible that we could decide to bless HTML outputs as able to execute JupyterLab commands, rather than this being an unintended side-effect.

**Pull request closed without merging**